### ISSUE TITLE
lein test-refresh doesn't run any tests

### ISSUE THREAD

**Issue by expez** [jakemcc/lein-test-refresh#35]
Hello!  

I realize you mostly hear from people when they have something to complain about so let me just start this off on a positive note and let you know that I love your plugin.  Development is a lot more enjoyable with instantaneous feedback!

Now, on this one project I get this output from `test-refresh`:

```
Ran 0 tests containing 0 assertions.
0 failures, 0 errors.
```

and it stubbornly sticks to that story.

The output from `lein test` is a lot more interesting:

```
Ran 12 tests containing 116 assertions.
0 failures, 0 errors.
```

AFAIK there's nothing special about this project, but then I'm not sure what might cause this.  I'm running `0.9.0` of `lein-test-refresh`

```
$ lein --version
Leiningen 2.5.1 on Java 1.8.0_45 OpenJDK 64-Bit Server VM
```

**Comment by jakemcc** [comment]
Glad to hear you enjoy it. Once you get used the quick feedback it is hard to go back.


Any chance the project is public? If not, are you able to recreate with a minimal project that could be shared? Definitely want to figure this out.

**Comment by jakemcc** [comment]
@expez Any more details? Could you share the project.clj?

**Comment by expez** [comment]
Sorry for going off the grid, but the project isn't public and I haven't had the time to try to cut it down to something minimal which reproduces the problem.

Do you have any idea as to where I might start looking?

**Comment by jakemcc** [comment]
If I were looking into it I'd start by looking at my project.clj and try to see what non-default options I'm using. You can sort of see what options test-refresh directly depends on by looking at test-refresh code. 

[This](https://github.com/jakemcc/lein-test-refresh/blob/master/test-refresh/src/leiningen/test_refresh.clj#L41-L44) is where test-refresh pulls out the `:test-paths`: option which tells it what directories tests are found in. You can see it also constructing map of other options from the project.clj in the `parse-commandline` function (which, when reading the code now, that function could clearly be better named).

It sort of seems like it isn't noticing your test files at all or some test selector is preventing it from running your tests.


If you have time to attempt to slim down the project to something you could at least share with me I'd be really happy to look at it. Perhaps try isolating a utility namespace and removing any private dependencies from the project.clj.

**Comment by expez** [comment]
Thanks for the tip, I found the problem pretty quickly after reading that.  I was putting some stuff I wanted shared across leiningen profiles in the `:base` profile.  AFAIK leiningen merges this into its own `:base` profile, which contains stuff like the default src and test dirs as well as sensible defaults for the jvm.  I haven't explicitly mentioned the source or test paths in my `project.clj`, so I think that means the `:base` profile isn't replaced but merged with the one in my `profiles.clj`.  I tried finding the base profile in the leiningen source to investigate further but I couldn't find it.

tl;dr, if you do `:profiles {:base {}}` `lein test-refresh` stops working but `lein test` still works.

**Comment by expez** [comment]
If [this](https://github.com/technomancy/leiningen/blob/ff84da697249184874b528950048981621ac0b61/leiningen-core/src/leiningen/core/project.clj#L478-L492) is the base profile then I'm not sure why setting `:base` to `{}` had any effect at all.


Inlined:
```clj
         :base {:resource-paths ["dev-resources"]
                :jvm-opts (with-meta tiered-jvm-opts
                            {:displace true})
                :test-selectors {:default (with-meta '(constantly true)
                                            {:displace true})}
                ;; bump deps in leiningen's own project.clj with these
                :dependencies '[[org.clojure/tools.nrepl "0.2.9"
                                 :exclusions [org.clojure/clojure]]
                                [clojure-complete "0.2.3"
                                 :exclusions [org.clojure/clojure]]]
                :checkout-deps-shares [:source-paths
                                       :test-paths
                                       :resource-paths
                                       :compile-path
                                       #'classpath/checkout-deps-paths]}
```

**Comment by jakemcc** [comment]
Hmm, yeah, that is interesting. I would have expected the different `:base` profiles to end up merged together using Leiningen's merging logic.

You might be able to use the `:provided` or `:user` profiles to set some global options.

Well, I'm going to close this. Thanks for digging into it. Glad to hear the problem is sort of solved (or at least known).

**Comment by expez** [comment]
Thanks for your help in tracking this down! :+1:
